From eb589e7b8d99dc05e2eea2223113f37c15e12cc7 Mon Sep 17 00:00:00 2001
From: Brian Behlendorf <behlendorf1@llnl.gov>
Date: Mon, 2 Nov 2015 16:55:17 -0800
Subject: [PATCH 2/3] Treat RETRY as FAILURE

When testing kernel modules with buildbot it's entirely possible that
the system may panic resulting in a disconnect.  For the purposes of
automated testing we want to consider this a FAILURE and immediately
move on to testing the next change.

Signed-off-by: Brian Behlendorf <behlendorf1@llnl.gov>
---
 master/buildbot/process/results.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/master/buildbot/process/results.py b/master/buildbot/process/results.py
index 9affa02..fac912e 100644
--- a/master/buildbot/process/results.py
+++ b/master/buildbot/process/results.py
@@ -51,7 +51,10 @@ def computeResultAndTermination(obj, result, previousResult):
             possible_overall_result = WARNINGS
         if obj.flunkOnWarnings:
             possible_overall_result = FAILURE
-    elif result in (EXCEPTION, RETRY, CANCELLED):
+    elif result == RETRY:
+        possible_overall_result = FAILURE
+        terminate = True
+    elif result in (EXCEPTION, CANCELLED):
         terminate = True
 
     result = worst_status(previousResult, possible_overall_result)
-- 
1.9.1

